Packs/TrustwaveFusion/Integrations/TrustwaveFusion/TrustwaveFusion.py

```diff
@@ -0,0 +1,587 @@
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    :type client: ``Client``
+    :param Client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ""
+    try:
+        client.describe()
+        message = "ok"
+    except DemistoException as e:
+        if e.res is not None and e.res.status_code == 401:
+            message = "Authorization Error: make sure API Key is correctly set"
+        elif "Forbidden" in str(e) or "Authorization" in str(e):
+            message = "Authorization Error: make sure API Key is correctly set"
+        else:
+            raise e
+    return message
+
+
+def fetch_incidents(client, max_results, first_fetch):
+    demisto.debug(f"Fetching {max_results}, {first_fetch} {type(first_fetch)}")
+    last_run = demisto.getLastRun()
+    last_fetch = last_run.get("last_fetch", None)
+    demisto.debug(f"last_run: {last_run}")
+    if last_fetch is None:
+        # if missing, use what provided via first_fetch
+        last_fetch = first_fetch
+    else:
+        # otherwise use the stored last fetch
+        last_fetch = int(last_fetch)
+
+    search_since = max(first_fetch, last_fetch)
+    demisto.debug(f"{first_fetch} <=> {last_fetch} == {search_since}")
+    created_since = timestamp_to_datestring(search_since * 1000, FUSION_DATE_FORMAT)
+
+    ticket_types = demisto.params().get("ticket_types")
+    incidents = []
+    try:
+        params = {
+            "pageSize": max_results,
+            "createdSince": created_since,
+            "sortField": "createdOn",
+            "sortDescending": "false",
+        }
+        if ticket_types:
+            params["type"] = ",".join(ticket_types)
+        demisto.debug(f"params: {params}")
+        tickets = client.search_tickets(**params)
+        latest_timestamp = search_since
+        if tickets:
+            demisto.debug(f"Found {len(tickets)} tickets from Fusion")
+            for tkt in tickets:
+                incident = {
+                    "name": tkt["subject"],
+                    "occurred": tkt["createdOn"],
+                    "description": tkt.get("description", ""),
+                    "severity": SEVERITY_MAP.get(tkt.get("priority"), 0),
+                    "rawJSON": json.dumps(tkt),
+                }
+                dt = dateparser.parse(tkt["createdOn"], settings={"TIMEZONE": "UTC"})
+                latest_timestamp = max(dt.timestamp(), latest_timestamp)
+                incidents.append(incident)
+    except DemistoException as e:
+        demisto.error(f"ERROR: {e}")
+        demisto.error(e.res)
+
+    # One second in the future to prevent duplicates
+    last_run = {"last_fetch": latest_timestamp + 1}
+    demisto.setLastRun(last_run)
+    demisto.incidents(incidents)
+
+
+def get_ticket_command(client, args):
+    ticket = client.get_ticket(args.get("id"))
```

Comment:
i assume that in case the ticket with the given id is not found, the function returns an error, which's not the wanted behavior, in such case we'd want to return a readable output that states so and not an error.